src/ARAnchorManager.js

```diff
@@ -0,0 +1,93 @@
+// import { EventEmitter } from 'events';
+import { Object3D, Vector3, Matrix4 } from 'three';
+/**
+ * A class to manage registration and unregistration of anchors
+ * for any THREE.Object3D. Anchors are world space and handle updating
+ * the position of the object to which one is assigned.
+ * AnchorManager emits events pertaining to anchors.
+ */
+export default class ARAnchorManager /* extends EventEmitter */ {
+  constructor(vrDisplay) {
+    // super();
+    if (!(vrDisplay instanceof VRDisplay) || !vrDisplay.getAnchors) {
+      throw new Error("invalid vr display parameter");
+    }
+    this.vrDisplay_ = vrDisplay;
+    this.anchorsToObject3Ds_ = new Map();
+    this.object3DsToAnchors_ = new Map();
+    this.vrDisplay_.addEventListener('anchorsupdated', 
+        event => this.onAnchorsUpdated_(event));
+    // This attribute is used to get the scale value from the anchor model
+    // matrix but not use it. It is just to avoid garbage collection.
+    this.scale_  = new Vector3();
+    // This attribute is used to calculate the matrix to be passed to the
+    // anchor creation code only using the position and quaternion from the
+    // object3d.
+    this.matrix_ = new Matrix4();
+    return this;
+  }
+  /**
+   * Assign a new anchor to an THREE.Object3D.
+   */
+  addAnchorForObject3D(object3d) {
+    if (!(object3d instanceof Object3D)) {
+      throw new Error('invalid object3d trying to add an anchor');
+    };
+    // Do not allow to create an anchor for an object3d that already has one.
+    if (this.object3DsToAnchors_.get(object3d)) {
+      return;
+    }
+    // Calculate the model matrix of the object3d using the position and
+    // quaternion of the object.
+    this.scale_.set(1, 1, 1);
+    this.matrix_.compose(object3d.position, object3d.quaternion, this.scale_);
```

Comment:
Deleted the comment. The problem is that we cannot user the scale factor of the object. At least to make sure about that, we are creating this new matrix as a precaution (and because is the same performance cost). I have also added some comments to the function so the caller knows that the object3d should be in world space. If he/she wants to use scale, at least it will be honored but not used.

Reply:
Sounds good -- in that case, can we add a console.warn/error when an object with a parent is used (maybe check if parent position/rotation !== identity)? Sometimes it could just be a group at 0, 0, 0 to make things easier, but that parent could potentially contain a transformation and we should display an error because it could be unintended